jobs/types/base_validation_errors.py

```diff
@@ -25,6 +25,12 @@
 import python_utils
 
 
+class ModelIdRegexException(Exception):
```

Comment:
Why not inherit from base audit error?

Reply:
There already exist a ModelIdRegexError which inherits from base audit error.Using that would make my function a generator. `_get_change_domain_class` returns with argument so it cannot be a generator

Reply:
I'm not sure I understand what you mean.

Reply:
if ModelIdRegexError inherits from base audit error, I will have to yield it, or return it. 
I can't yield it because then `_get_change_domain_class` will become generator. If I return it, it returns as a class instance
but if I inherit from Exception, I can simply raise it.
Does this makes sense ?

Reply:
Oh, right, is it possible to create both yieldable and raisable error class? cc @brianrodri

Reply:
Hmm I think it might make more sense to return None here. If the model ID doesn't match the regex, there's already a check that would catch that, so it should be fine to just "skip" this item when the ID doesn't match. Would that help?

Reply:
I guess that makes sense, thanks for explaining, updated, PTAL!

Reply:
Is the ModelIdRegexException being used?

Reply:
+1, seems like we can delete this class now.